### DISCUSSION TITLE
Verify iframe failed to load

### DISCUSSION THREAD

**Discussion by loihp** [WalletConnect/walletconnect-monorepo#3499] [Developer Support]
I got error when pop up modal qr showing
how to fix this, tks all !!!
<img width="565" alt="Screenshot 2023-08-18 at 16 09 55" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/67997721/dce2ee6b-e67b-46ed-a5ed-e40bdd669086">

<img width="578" alt="Screenshot 2023-08-18 at 16 37 05" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/67997721/93691b50-210c-4dbd-8b62-96fa3300d61c">

**Comment by glitch-txs** [comment]
This is being fixed, shouldn't affect the behaviour of WalletConnect's functionalities

**Reply by devon-n** [reply]
I have the same issue but it is affecting WalletConnect behaviour
```
provider = await UniversalProvider.init({
  projectId: process.env.REACT_APP_WALLET_CONNECT_PROJECT_ID,
  chains: [chainId],
  showQrModal: false,
  methods: ['personal_sign'],
  events: ['chainChanged', 'accountsChanged'],
});
await provider.connect({
  pairing: walletContext.walletProvider.split('/')[2]
})
await provider.enable();
```

**Reply by VasilPVR** [reply]
Where it was fixed? Can you point me to the link?

**Reply by jonandgon** [reply]
Could this bug be related to why I am getting these 404s? I'm using `@walletconnect/sign-modal-react` v2.6.1

<img width="966" alt="CleanShot 2023-08-25 at 10 53 27@2x" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/75697675/5ede05a0-7c8b-4437-9db5-d3573233300e">

**Comment by domid** [comment]
Having the same issue with new nextjs 13 app router based web app. Can't get it to work on any new nextjs web app. Can we get an eta on the fix?

<img width="1500" alt="Screenshot 2023-08-26 at 16 55 12" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/464714/779d9704-b930-43ac-845d-1a64c7151dcc">

**Comment by domid** [comment]
I believe this may be your walletconnect CORS config issue. I'm hoping it can get fixed in the next few days. We are blocked by this error.

**Reply by boidushya** [reply]
Hey could you elaborate how you're blocked by this error? Theoretically this shouldn't affect anything other than validation for Verify API

**Reply by domid** [reply]
@boidushya At this stage, we're facing concerns related to user experience and brand integrity. Launching a product with console warning errors, especially those suggesting potential issues with web3 security, could alarm users and erode trust. Additionally, I'm uncertain about the display of these console errors across different browsers, which adds another layer of uncertainty.

**Reply by boidushya** [reply]
You could verify your domain to not see these errors https://docs.walletconnect.com/2.0/web3wallet/verify

**Reply by devon-n** [reply]
We are blocked as we try to use Universal Provider with ethers but this error arises and gives us no provider to pass into ethers

**Reply by glitch-txs** [reply]
hi @devon-n could you share an [MRE](https://github.com/orgs/WalletConnect/discussions/3457)? Would like to take a look. Also this should be a non blocking issue so the error could be somewhere else I think 🤔

**Reply by devon-n** [reply]
Hi @glitch-txs,

Here is the code

This is run after the user has connected through SignClient. We use universal provider to then call contract functions etc.

Anything after UniversalProvider.init() does not run but I do get the verify error in the console

```
async function getProvider() {

    let provider = window.ethereum

    if (isMetaMaskProvider()) {
        //Make sure to get metamask provider
        if (window.ethereum?.providers?.length) {
            provider = window.ethereum.providers.find((provider) => provider.isMetaMask);
        }
    } else if (isWalletConnectProvider()) {
        provider = await UniversalProvider.init({
            projectId: process.env.REACT_APP_WALLET_CONNECT_PROJECT_ID,
            relayUrl: 'wss://relay.walletconnect.com',
            logger: 'info',
            namespaces: {
                eip155: {
                    methods: ['personal_sign'],
                    chains: [chainId],
                    events: ['chainChanged', 'accountsChanged']
                }
            }
        });
        await provider.connect({
            topic: walletContext.walletProvider.split('/')[2],
            namespaces: {
                eip155: {
                    methods: ['personal_sign'],
                    chains: [chainId],
                    events: ['chainChanged', 'accountsChanged']
                }
            }
        })
        await provider.enable();
        provider.on("session_delete", ({ id, topic }) => {
            walletContext.clearPgWallet()
        });
    } else if (isCoinBaseWalletProvider()) {
        provider = cbEthereum;
    }

    return provider;
}


export const getBlockchainContract = async (contractAddress, abi) => {
    try {
        let provider = await getProvider();

        let prov = new ethers.providers.Web3Provider(provider);

        let sign = prov.getSigner();

        return new ethers.Contract(contractAddress, abi, sign);
    } catch (err) {
        console.error("Blockchain Contract error: " + err);
    }
}

```

**Comment by kramlj00** [comment]
Hi, has anyone been able to solve this problem?

**Comment by zagambill** [comment]
Any update on the ETA for this fix? This does seem to be the issue breaking our implementation of Rainbowkit's NextAuth SIWE -- we're stuck on walletconnect v1 because upgrading rainbowkit-siwe/WCv2 seems to require this endpoint, but with this bug users cannot sign in at all: they get all the way to signing the authentication message, but then rainbowkit-siwe/nextauth is unable to verify so cannot authenticate the user session. 

The dapp is peer-to-peer token safe swapping so SIWE is a must, and we'd prefer to keep using rainbowkit
